frmts/vrt/pixelfunctions.cpp

```diff
@@ -1165,6 +1222,99 @@ CPLErr InterpolatePixelFunc( void **papoSources, int nSources, void *pData,
     return CE_None;
 }
 
+static const char pszNanPixelFuncMetadata[] =
+"<PixelFunctionArgumentsList>"
+"   <Argument type='builtin' value='NoData'>"
+"   </Argument>"
+"</PixelFunctionArgumentsList>";
+
+static CPLErr NanPixelFunc( void **papoSources, int nSources, void *pData,
```

Comment:
a more generic function would be ReplaceNoData() that would take a OutputNoData value that could be nan